src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -489,7 +527,10 @@ public void Log(ref LogEntry entry, TaskCompletionSource<object> tcs = null)
                 state = _localState.Value = GenerateThreadWriterState();
             }
 
-            if (state.Free.Dequeue(out var item))
+            int currentProcessNumber = JsonContextPoolHelpers.CurrentProcessNumber;
+            var pool = _pooledMessageEntries[currentProcessNumber% _pooledMessageEntries.Length];
```

Comment:
Why do we need `% _pooledMessageEntries.Length` ? Shouldn't property `CurrentProcessNumber` already ensure that value is adjusted to `Environment.ProcessorCount`?

Reply:
We technically don't need it, but I want to be on the safe side here.

Reply:
Maybe it is better to do it inside the property then?

Reply:
Moved it so it is only called once.